Competitive_Programming/Python/DP/Binomial_Coefficient/binomial_coefficient.md

```diff
@@ -0,0 +1,103 @@
+# Binomail Coefficient Using Bottom-Up Tabularization:
+
+Here, firstly we will see the conventional method to obtain the binomial coefficient and how we can reduce the complexity of it using bottom-up tabularization.
+
+The expression to obtain binomial coefficient is given as:
+
+<img src='https://user-images.githubusercontent.com/65852362/142717568-8816fda5-63af-4e44-88fd-1cbf7eed6f1e.png'>
+
+## Algorithm to obtain binomial coefficient:
+
+```
+function C(n, k)
+    if k=0 or k=n
+        then return 1
+    else 
+        return C(n-1, k-1)+ C(n-1, k)
+```
+
+## Time Complexity Analysis:
+
+The time complexity of the above algorithm will be <b>O(2<sup>n</sup>)</b>. Here, this much high time complexity is not feasible, so to reduce the complexity, we will store the value of each computation in a tabular form.
+
+## Bottom-Up Tabularization Approach:
+
+Here, <b>bottom-up approach means to first solve smaller subproblems and then to use the solution of those smaller subproblems to get the solution of larger subproblems.</b>
+
+We will also form a table to obtain the value of C(n, k) whose columns will represent the k-value and rows will represent the n-values. 
+
+<b>C(n,k) returns the number of ways to form an unordered collection of k-items selected from a collection of n-distinct items.</b>
+
+Consider the below table for calculation of C(n, k):
+
+<img src='https://user-images.githubusercontent.com/65852362/142717582-551212e7-2bc2-4a62-9ebe-66c6f236b5d4.png'>
+
+Here, if k = 0 or k = n, we will store value '1' in the cell corresponding to that n and k values.
+
+<img src='https://user-images.githubusercontent.com/65852362/142717592-7d210bc6-8494-4dcd-9cb6-2baf79f7a520.png'>
+
+Now, for other cells, having n-value as 'i' and k-value as 'j', where <b>j != 0 and j != i</b>, C(i,j) will be the sum of <b>C(i-1, j-1)</b> and <b>C(i-1, j)</b>.
+
+For example, 
+```
+C(2,1) = C(1,1) + C(2, 0)
+C(2, 1) = 1 + 1
+C(2, 1) = 2
+```
+<img src = '(https://user-images.githubusercontent.com/65852362/142717603-5b95a41f-ab49-4347-9210-47ff9aa3d4c4.png'>
```

Comment:
add width and height for the images, so that it can fit nicly to the doc and easy to read for users